runtime/src/lib.rs

```diff
@@ -14,12 +14,18 @@ pub use writeable::DiplomatWriteable;
 mod result;
 pub use result::DiplomatResult;
 
+/// Like [`char`], but unvalidated.
 pub type DiplomatChar = u32;
 
+/// Like [`str`], but unvalidated.
 pub type DiplomatStr = [u8];
 
+/// Like `Wstr`, but unvalidated.
 pub type DiplomatStr16 = [u16];
 
+/// Like [`u8`], but interpreted explicitly as a raw byte as opposed to a numerical value.
```

Comment:
nit: more docs please, explain that this is only relevant for backends (like Dart) where a list of integers is different from a byte buffer